main.go

```diff
@@ -189,68 +73,72 @@ func main() {
 				}
 				cancel()
 
-				isCacherAvailableMu.Lock()
-				isCacherAvailable = isCacherAvailableTemp
-				isCacherAvailableMu.Unlock()
+				httpserver.IsCacherAvailableMu.Lock()
+				httpserver.IsCacherAvailable = isCacherAvailableTemp
+				httpserver.IsCacherAvailableMu.Unlock()
```

Comment:
This should be done through a Setter function, `httpserver.SetCacherAvailable(isCacherAvailableTemp)`